org.envirocar.app/src/org/envirocar/app/views/recordingscreen/TrackMapFragment.java

```diff
@@ -296,59 +296,59 @@ protected void injectDependencies(BaseApplicationComponent baseApplicationCompon
     @SuppressWarnings( {"MissingPermission"})
     private void enableLocationComponent(@NonNull Style loadedMapStyle) {
         // Check if permissions are enabled and if not request
-        if (PermissionsManager.areLocationPermissionsGranted(getContext())) {
-
-            // Get an instance of the component
-            locationComponent = mapboxMap.getLocationComponent();
-
-            // Activate with options
-            locationComponent.activateLocationComponent(
-                    LocationComponentActivationOptions
-                            .builder(getContext(), loadedMapStyle)
-                            .useDefaultLocationEngine(true)
-                            .locationEngineRequest(new LocationEngineRequest.Builder(750)
-                                    .setFastestInterval(750)
-                                    .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
-                                    .build())
-                            .build());
-
-            // Enable to make component visible
-            locationComponent.setLocationComponentEnabled(true);
-
-            // Set the component's camera mode
-            locationComponent.setCameraMode(CameraMode.TRACKING);
-
-            // Set the component's render mode
-            locationComponent.setRenderMode(RenderMode.COMPASS);
-        } else {
-            permissionsManager = new PermissionsManager(this);
-            permissionsManager.requestLocationPermissions(getActivity());
-        }
+//        if (PermissionsManager.areLocationPermissionsGranted(getContext())) {
+//
+//            // Get an instance of the component
+//            locationComponent = mapboxMap.getLocationComponent();
```

Comment:
While refactoring, it came to my knowledge that we utilize some location APIs from Mapbox inside recording screen. It will be interesting to see how our `org.envirocar.map` module will take care of this.

Probably building some provider independent implementation to update map camera & current location marker based on location updates will be a good idea, by accessing GPS location directly from Android's [`LocationManager`](https://developer.android.com/reference/android/location/LocationManager).

Reply:
Oh yes, good point. In any case, we should implement a provider independent approach for location updates. Unfortunately, implementing such an abstraction will not be straightforward since the MapBox `LocationComponent` seems to have a direct interaction between location updates and camera animations.

If decide to introduce Android's LocationManager we explicitly have to implement some event listener routines that listens to location updates and triggers the camera updates. Another solution would be, to introduce an own `LocationComponent` that encapsulates the interaction between location and camera updates. We could have different implementations then, e.g. based on provider specific implementations such as the one in MapBox.

I'd suggest, to review all the possible map provider libs, first, to see if other libs such as MapLibre or OSMDroid also have some provider specific routines for handling location updates. Based on your review, we should decide which way to go.